src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -665,31 +665,47 @@ private void addEnrollmentsToSession(HttpServletRequest request,
 		M_log.debug("ENROLLMENTS: " + sb.toString());
 		HashMap<Integer, String> enrollmentsFound =  new HashMap<Integer, String>();
 		JSONArray enrollmentsArray = new JSONArray(sb.toString());
+
+		//iterate through new enrollments
+		//if enrollment type == teacherEnrollment or DesignerEnrollment with Designer Role then add it to enrollments found and break
+		//Else enrollment type is TAEnrollment
+		//This is because Teachers and Designers can add friends, but not Librarians (a type of Designer)
 		for(int i = 0; i < enrollmentsArray.length(); i++){
 			JSONObject childJSONObject = enrollmentsArray.getJSONObject(i);
 			M_log.debug("ENROLLMENT RECORD: " + childJSONObject.get("course_id") + " " + childJSONObject.get("course_section_id") + " " + childJSONObject.get("type"));
-			//if the key is already in the map, check and see if the new value is greater for that key
-			//if it is greater then replace, if not then skip.
-
-			if(enrollmentsFound.containsKey(childJSONObject.getInt("course_id"))){
-				String oldType = enrollmentsFound.get(childJSONObject.getInt("course_id"));
-				String newType = childJSONObject.getString("type");
-				M_log.debug("OLD TYPE: " + oldType);
-				M_log.debug("NEW TYPE: " + newType);
-				//Only want to overwrite the key,value pair if the value is greater
-				if(enrollmentsMap.get(newType) > enrollmentsMap.get(oldType)){
-					enrollmentsFound.put(childJSONObject.getInt("course_id"), childJSONObject.getString("type"));
-				}
+
+			String newType = childJSONObject.getString("type");
+			String newRole = childJSONObject.getString("role");
+
+			M_log.debug("NEW TYPE: " + newType);
+			M_log.debug("NEW ROLE: " + newRole);
+
+			boolean isTeacher = false;
+			boolean isDesigner = false;
+			
+			if(newType.equals("DesignerEnrollment") && newRole.equals("DesignerEnrollment")){
```

Comment:
For designer we need to check both because a type designer with a role of designer can add anyone, but a designer type with role other than designer can only add students and observers.